person.rb

```diff
@@ -1,23 +1,56 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
 class Person
   attr_reader :id
   attr_accessor :name, :age, :parent_permission
 
-  def initialize(age:, name: 'unknown', parent_permission: true)
+  def initialize(*args)
     @id = Random.rand(1..1000)
-    @name = name
-    @age = age
-    @parent_permission = parent_permission
+    @age = args[0]
+    @name = args[1] || 'unknown'
+    @parent_permission = args[2].nil? ? true : args[2]
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age?
     @age >= 18
   end
 end
 
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
 
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name()
+    @nameable.correct_name.upcase
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0, 10]
+  end
+end
```

Comment:
- [OPTIONAL] If we are going the `Ruby way`, then it will be nice to put each class in a separate file, please kindly fix that. If you place multiple classes in the same file and they start to grow, then you're going to need to separate them later. This [article](https://stackoverflow.com/questions/6776897/multiple-classes-in-one-file-ruby-style-question#:~:text=If%20you%20configure%20your%20own,for%20more%20advanced%20design%20patterns.) should throw more light on that.